site/.eleventy.js

```diff
@@ -0,0 +1,181 @@
+  rpi4: "Raspberry Pi 4 (32-bit)",
+  "rpi4-64": "Raspberry Pi 4",
+  tinker: "ASUS Tinker Board",
+  yellow: "Home Assistant Yellow",
+};
+
+const randomColor = () => `#${Math.floor(Math.random() * 16777215).toString(16)}`
+const dataPoint = (data) => ({ borderWidth: 1, pointRadius: 0, fill: false, borderColor: randomColor(), ...data })
+const sortTableData = (tableData) => tableData.sort((a, b) => b.installations - a.installations).map((entry, idx) => ({ ...entry, idx: idx + 1 }));
+
+module.exports = function (eleventyConfig) {
+  eleventyConfig.addPassthroughCopy({ "src/_static": "static" });
+
+  eleventyConfig.addFilter("historyFiltering", (history) => historyFiltering(history))
+  eleventyConfig.addFilter("countriesForMap", (base) => JSON.stringify(Object.keys(base || {}).reduce((obj, key) => ({ ...obj, [key]: { installations: base[key] || 0 } }), {})));
+  eleventyConfig.addFilter("calculatePercentage", (total, part, decimal) => calculate.percentage(total, part, decimal));
+  eleventyConfig.addFilter("formatNumber", (value) => Intl.NumberFormat().format(value));
+  eleventyConfig.addFilter("keys", (data) => Object.keys(data));
+  eleventyConfig.addFilter("values", (data) => Object.values(data));
+
+  eleventyConfig.addFilter("versionSummary", (data) => {
+    const releases = Object();
+    const results = {}
+
+    Object.keys(data).forEach((version) => {
+      if (!version.includes("dev")) {
+        const key = version.split(".").slice(0, 2).join(".");
+        releases[key] = (releases[key] || 0) + data[version];
+      }
+    });
+
+    const allRows = Object.keys(releases)
+      .filter((key) => releases[key] > 100)
+      .sort((a, b) => {
+        const mainVersionCmp =
+          parseInt(b.split(".")[0]) - parseInt(a.split(".")[0]);
+        if (mainVersionCmp !== 0) {
+          return mainVersionCmp;
+        }
+        return parseInt(b.split(".")[1]) - parseInt(a.split(".")[1]);
+      })
+
+
+    allRows.slice(0, 5).forEach(version => { results[version] = releases[version] })
+
+    if (allRows.length > 5) {
+      results["Other"] = allRows
+        .slice(5)
+        .reduce((accumulator, item) => accumulator + releases[item], 0)
+    }
+
+    return results
+  });
+
+  eleventyConfig.addFilter("osVersionSummary", (data) => {
+    const results = {}
+
+    const sortedVersions = Object.keys(data).sort((a, b) => data[b] - data[a]);
+    sortedVersions.slice(0, 5).forEach(version => { results[version] = data[version] })
+
+    if (sortedVersions.length > 5) {
+      results["Other"] = sortedVersions
+        .slice(5)
+        .reduce((accumulator, currentValue) => accumulator + data[currentValue], 0)
+    }
+
+    return results
+  });
+
+  eleventyConfig.addFilter("sortBoards", (value) => {
+    const data = { data: [], labels: [], backgroundColor: [] }
+    Object.keys(value).sort().forEach(key => { data.labels.push(friendlyBoardName[key] || key); data.data.push(value[key]); data.backgroundColor.push(randomColor()) })
```

Comment:
let's give this some prettier love

Reply:
btw alternative for forEach that I think is slightly more readable:

```js
for (const key of Object.keys(value).sort()) {
  data.labels.push(friendlyBoardName[key] || key);
  data.data.push(value[key]);
}
```

Reply:
Yeah, this was just copy+paste from old site